pkg/archive/archive.go

```diff
@@ -530,7 +550,8 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
 		}
 
 	case tar.TypeLink:
-		targetPath := filepath.Join(extractDir, hdr.Linkname)
+		lpath := adjustHardLink(hdr.Linkname, hdr.Name)
+		targetPath := filepath.Join(extractDir, lpath)
```

Comment:
> Is `Replace()` safe to use here?

I'm pretty sure that in this code path, `oldBase` is guaranteed to be at the beginning of `hdr.Name` but your suggestion is definitely safer. I may be overlooking something, but it should be fine as long as all of the existing CLI tests around `docker cp` still pass.